base/multidimensional.jl

```diff
@@ -1496,3 +1496,173 @@ end
 function Base.showarg(io::IO, r::Iterators.Pairs{<:CartesianIndex, <:Any, <:Any, T}, toplevel) where T<:AbstractVector
     print(io, "pairs(IndexCartesian(), ::$T)")
 end
+
+## sortslices
+
+"""
+    sortslices(A; dims, alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
+
+Sort slices of an array `A`. The keyword argument `dims` must be either
+an integer or a tuple of integers specifies along which dimension to take the
+slices.
+
+If `A` is a matrix, `dims=1` will sort rows, `dims=2` will sort columns.
+Note that the default comparison function on one dimensional slices sorts
+lexicographically.
+
+For convenience, `dims=:rows` and `dims=:cols` are supported as alternative
```

Comment:
Can we get rid of these awkward special cases?

Reply:
Agreed; this makes supporting named dimensions a breaking change, so we shouldn't have this now.